optaplanner-core/src/main/java/org/optaplanner/core/impl/score/definition/ScoreDefinition.java

```diff
@@ -67,6 +67,22 @@
      */
     S getZeroScore();
 
+    /**
+     * @param score never null
+     * @return true if the score is higher or equal to {@link #getZeroScore()}
+     */
+    default boolean isPositive(S score) {
+        return score.compareTo(getZeroScore()) >= 0;
+    }
+
+    /**
+     * @param score never null
+     * @return true if the score is lower or equal to {@link #getZeroScore()}
+     */
+    default boolean isNegative(S score) {
+        return score.compareTo(getZeroScore()) <= 0;
```

Comment:
In arithmetic, zero is both positive and negative, but neither strictly positive or strictly negative. OptaPlanner follows those conventions.

Score weights are positive, which includes zero to turn that constraint off.

Reply:
In [arithmetic](http://mathworld.wolfram.com/Arithmetic.html), positive integers are {1, 2, 3, ...}. Zero is [neither negative nor positive](http://mathworld.wolfram.com/Zero.html). Diverging from this convention may be very confusing. I mean, I can imagine that in score context it might be more convenient to bend the definition slightly but I fear that for most people the terms are so deeply rooted that this would only raise confusion.

Let's take a look at the only usage of the `isPositive()` method:
```java
if (!getScoreDefinition().isPositive(constraintWeight)) {
    throw new IllegalArgumentException("The constraintWeight () for constraintPackage () \
            and constraintName () of constraintConfigurationClass () must have \
            a positive or zero constraintWeight ().");
```
Even the exception message suggests that "positive"  means strictly positive. Additionally, if we correct the logic of the the two methods the piece of code becomes even clearer:
```java
if (getScoreDefinition().isNegative(constraintWeight)) {
    throw new IllegalArgumentException("The constraintWeight () ... must have \
            a positive or zero constraintWeight ().");
```
I strongly recommend to change the method name (or the condition) to be consistent with the mathematical definition (e.g. `isNonNegative()` and `isNonPositive()` and also to be consistent with usage and understanding of the terms "positive" and "negative" in the rest of the code and its documentation.

Reply:
"Is 0 A positive number? It really depends on context. In common use in English language, zero is unsigned, that is, it is neither positive nor negative. In typical French mathematical usage, zero is both positive and negative."

This is a very great good catch, thanks for pointing it out. I 'll rename the method to avoid any ambiguity
 isPositiveOrZero()
 isStrictlyPositive()
 the naming isPostive() is ambiguous and therefore banned